snippets/card-collection.liquid

```diff
@@ -4,7 +4,8 @@
   Accepts:
```

Comment:
To reply to your comment @Roi-Arthur 

I think we can change the logic a little. the `desktop` variable is assigned `at_least: 1` so in some cases we don't need to assign anything in the collage file. 
Here is the mobile logic you have but a bit reworked, I'd have to actually test it to make sure the `and` is read properly:
```liquid
{% liquid 
  unless section.settings.mobile_layout == 'column' or section.block.size == 1
    if section.settings.mobile_layout == 'collage'
      if section.blocks.size == 3 and forloop.first and section.settings.desktop_layout == left or forloop.last and section.settings.desktop_layout == right
        assign columns_amount_mobile = 1
      else
        assign columns_amount_mobile = 2
      endif
    endif
  endunless
%}
```

For the complementary products it's a bit trickier 🤔 I think from what I can see their size is going from `64px` wide up to potentially `140px` when it's on a product with no image. And `118px` on a product with images. 

Maybe we could do something a bit simpler that isn't quite perfect but good enough. Something like defaulting to `10` for desktop and `5` for mobile 👍 This way we don't go into logic that's a bit hard to understand. 

What do you think?

Reply:
> the desktop variable is assigned at_least: 1 so in some cases we don't need to assign anything in the collage file.

I did think of this after the fact and was wondering what's preferable: one the one hand we have more liquid but it's very readable and you know exactly what you're passing to the snippet so might be more user friendly, especially for people who are getting started. On the other we can definitely remove all cases where the assign is for 1 in terms of efficiency.

> I'd have to actually test it to make sure the and is read properly:

That was also my concern 😅
Haven't tried any of this yet; was just a draft starting from what sounded logical.

Reply:
Just a correction, you mentioned that `operators are evaluated from left to right` which isn't right. It's the opposite. 

![](https://screenshot.click/10-21-q1hd7-obzp1.png)